esmvaltool/diag_scripts/seaice/seaice_sensitivity.py

```diff
@@ -0,0 +1,343 @@
+        f"None or too many matching files found for {variable_group}"
+    )
+
+    # Load the cube, [0] is because selection returns a list
+    cube = iris.load_cube(selection[0]["filename"])
+
+    return cube
+
+
+def calculate_regression(independent, dependent):
+    """Use SciPy stats to calculate the least-squares regression."""
+    logger.debug(
+        "Calculating linear relationship between %s and %s",
+        dependent,
+        independent,
+    )
+
+    # Use SciPy stats to calculate the regression
+    # result = slope, intercept, rvalue, pvalue, stderr
+    result = stats.linregress(independent, dependent)
+
+    # Return everything
+    return result
+
+
+def calculate_annual_trends(data):
+    """
+    Calculate annual trends for surface air temperature (tas) and sea ice area (siconc).
+
+    Also used for the r and p values from the regression of siconc as a function of tas.
+    """
+    logger.debug("calculating annual trends")
+
+    # Load the preprocessed cubes
+    si_cube = extract_cube(data, "siconc")
+    tas_cube = extract_cube(data, "tas")
+
+    # Calculate the individual trends over time
+    years = tas_cube.coord("year").points
+    si_trend = calculate_regression(years, si_cube.data)
+    tas_trend = calculate_regression(years, tas_cube.data)
+
+    # Calculate the direct regression for r and p values
+    direct_regression = calculate_regression(tas_cube.data, si_cube.data)
+
+    dictionary = {
+        "si_ann_trend": si_trend.slope,
+        "tas_ann_trend": tas_trend.slope,
+        "direct_r_val": direct_regression.rvalue,
+        "direct_p_val": direct_regression.pvalue,
+    }
+
+    return dictionary
+
+
+def calculate_direct_sensitivity(data):
+    """Calculate slope of sea ice area over global mean temperature."""
+    logger.debug("calculating sensitivity")
+
+    # Load the preprocessed cubes
+    si_cube = extract_cube(data, "siconc")
+    tas_cube = extract_cube(data, "tas")
+
+    # Calculate the slope of the direct regression, NOT via time
+    sensitivity = calculate_regression(tas_cube.data, si_cube.data)
+
+    return sensitivity.slope
+
+
+def create_titles_dict(data):
+    """
+    Create a dictionary of appropriate observations and titles.
+
+    Values depend on whether the plot is for the Arctic or Antarctic
+    and assume the recipe used September Arctic sea ice data or
+    annually mean averaged Antarctic sea ice data
+    """
+    dictionary = {}
+
+    first_variable = next(iter(data))
+
+    if first_variable["diagnostic"] == "arctic":
+        # Ed Blockley's dSIA/dT values in million sq km per Kelvin
+        dictionary["obs"] = {
+            "1979-2014": {"mean": -4.01, "std_dev": 0.32, "plausible": 1.28}
+        }
+        # Setting titles for plots
+        dictionary["titles"] = {
+            "notz_fig_title": "September Arctic Sea Ice Sensitivity",
+            "notz_ax_title": f"dSIA/dGMST obs from {next(iter(dictionary['obs'].keys()))}",
```

Comment:
```suggestion
            "notz_fig_title": "September Arctic sea-ice area sensitivity to global mean surface temperature",
            "notz_ax_title": f"Annual regression over {next(iter(dictionary['obs'].keys()))}",
```
Suggested more verbose titles. 

I removed "obs" - is that OK or does it obscure an intended meaning?

Reply:
I think yours is better, but only the (currently hardcoded observations are from those years, the years of the data are configurable in the recipe (although they are currently set to the same), so I probably need a different title.